native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -1232,23 +1343,30 @@ static void update_workers_lbstatus(proxy_server_conf *conf, apr_pool_t *pool, s
                 rnew->headers_in = apr_table_make(rnew->pool, 1);
                 rv = proxy_cluster_try_pingpong(rnew, worker, url, conf, ou->mess.ping, ou->mess.timeout);
 
-                if (read_node_worker(id[i], &ou, worker) != APR_SUCCESS)
+                ap_assert(node_storage->lock_nodes() == APR_SUCCESS);
+                if (read_node_worker(id[i], &ou, worker) != APR_SUCCESS) {
+                    /* the node is gone or something like that */
+                    node_storage->unlock_nodes();
                     continue;
+                }
 
                 if (rv != APR_SUCCESS) {
                     /* We can't reach the node: XXX changing ou->mess.updatetimelb here ??? */
+                    /* XXX if this is a timeout, we have might have a outdated list of nodes!!! */
```

Comment:
Typos: `we have might have` -> `we might have` and `a outdated` -> `an outdated`.

Reply:
fixed